Networking/Sources/Networking/Networking.swift

```diff
@@ -1,6 +1,49 @@
+import Alamofire
+import Foundation
+
 public struct Networking {
     public private(set) var text = "Hello, World!"
 
     public init() {
     }
+    
+    public func perform<M: Decodable, T: TargetType>(target: T, responseClass: M.Type, completion: @escaping (Result<M?, NSError>) -> ()){
+        let method = Alamofire.HTTPMethod(rawValue: target.method.rawValue)
+        let headers = Alamofire.HTTPHeaders(target.headers ?? [:])
+        let params = buildParams(task: target.task)
+        
+        
+        AF.request(target.baseURL + target.path, method: method, parameters: params.0,encoding: params.1, headers: headers).responseJSON { (response) in
+            guard let statusCode = response.response?.statusCode else{
+                completion(.failure(NSError()))
+                return
+            }
+            
+            if statusCode == 200 {
```

Comment:
All status code from 200 ..< 300 are considered as success status codes

Reply:
Alamofire has a handler to validate success status codes which could be handled.. It's implemented by default I think.